sdks/java/core/src/main/java/org/apache/beam/sdk/io/TextIO.java

```diff
@@ -946,17 +987,22 @@ private boolean tryToEnsureNumberOfBytesInBuffer(int minCapacity) throws IOExcep
   @VisibleForTesting
   static class TextSink<T> extends FileBasedSink<T> {
     private final Coder<T> coder;
+    private final String header;
```

Comment:
These will need to be the encoded versions of the values stored as byte[]s since T may not be java serializable